### ISSUE TITLE
Crash when toggling fullscreen (Windows)

### ISSUE THREAD

**Issue by SigmaVirus** [loganmc10/GLupeN64#51]
This one is easy to reproduce on Windows and it doesn't matter which settings you use either.
Load a game, press F to go fullscreen and than BAAM it crashes.

**Comment by loganmc10** [comment]
I can reproduce this, I need to get apitrace working on Windows in order to figure out where it is crashing.

**Comment by loganmc10** [comment]
I believe this is because ```context_reset``` isn't being called during the toggle, but it is in Linux for some reason, need to investigate further.

**Comment by loganmc10** [comment]
This same thing happens in mupen64plus-libretro by the way (Glide64 just gives you a black screen, gln64 crashes)

**Comment by loganmc10** [comment]
I logged an issue with RetroArch https://github.com/libretro/RetroArch/issues/3549

**Comment by SigmaVirus** [comment]
Thanks for all your help logan!

**Comment by orbea** [comment]
I can reproduce this in Slackware64-current, it does not happen with mupen64plus-libretro.

`RetroArch-2016.09.11_ebeb081_master-x86_64-1_git`
`GLupeN64-2016.09.09_28f8ae4_master-x86_64-1_git`
`mesa-2015.09.17_148fbf3_master-x86_64-1_git`(Recent commit despite the date)

```
Mesa: User error: GL_INVALID_OPERATION in glBindVertexArray(non-gen name)
Mesa: User error: GL_INVALID_OPERATION in glBindBuffer(non-gen name)
Mesa: User error: GL_INVALID_OPERATION in glVertexAttribPointer(no array object bound)
Mesa: 4 similar GL_INVALID_OPERATION errors
Mesa: User error: GL_INVALID_OPERATION in glBindFramebuffer(buffer)
Mesa: User error: GL_INVALID_VALUE in glUseProgram
Mesa: User error: GL_INVALID_OPERATION in glBindTexture(non-gen name)
Mesa: User error: GL_INVALID_VALUE in glDeleteProgram
Mesa: User error: GL_INVALID_VALUE in glDeleteProgram
Mesa: User error: GL_INVALID_VALUE in glDeleteShader
Mesa: User error: GL_INVALID_VALUE in glDeleteShader
Mesa: User error: GL_INVALID_VALUE in glDeleteShader
Mesa: User error: GL_INVALID_VALUE in glDeleteShader
Mesa: User error: GL_INVALID_VALUE in glDeleteShader
Mesa: User error: GL_INVALID_VALUE in glDeleteShader
Mesa: User error: GL_INVALID_VALUE in glDeleteShader
Mesa: User error: GL_INVALID_VALUE in glDeleteShader
Mesa: User error: GL_INVALID_VALUE in glDeleteShader
Mesa: User error: GL_INVALID_VALUE in glDeleteProgram
Mesa: User error: GL_INVALID_VALUE in glDeleteProgram
Mesa: User error: GL_INVALID_VALUE in glDeleteProgram
Mesa: User error: GL_INVALID_VALUE in glDeleteProgram
Mesa: User error: GL_INVALID_VALUE in glDeleteProgram
Mesa: User error: GL_INVALID_VALUE in glDeleteProgram
Mesa: User error: GL_INVALID_VALUE in glDeleteProgram
Mesa: User error: GL_INVALID_VALUE in glDeleteProgram
OpenGL Error: invalid operation
retroarch: GLideN64/src/Textures.cpp:505: void TextureCache::init(): Assertion `!isGLError()' failed.
context_reset.
plugin_start_gfx success.
context_reset.

Thread 1 "retroarch" received signal SIGABRT, Aborted.
0x00007ffff1cb75cf in raise () from /usr/lib64/debug/libc.so.6
```

Full gdb log: http://pastebin.com/FrxK3buT

**Comment by loganmc10** [comment]
@orbea can you provide me an apitrace of when it happens?

**Comment by orbea** [comment]
Additionally this only occurs with `DEBUG=1`.

**Comment by orbea** [comment]
Apitrace: http://ks392457.kimsufi.com/orbea/stuff/trace/libretro/glupen64.trace.xz

**Comment by orbea** [comment]
I also checked the test core twinaphex linked in the RetroArch issue, it does not occur there.

**Comment by loganmc10** [comment]
@orbea I believe I fixed the issues on Linux with 189c53cf00f5dcbe17ecb8bae69f4dde0ca7b16d, but it is still crashing in Windows.

**Comment by orbea** [comment]
Yea, thanks! I can confirm both the crash and mesa user errors are resolved on linux.

**Comment by loganmc10** [comment]
@SigmaVirus This should be fixed now (6d58b080136fd80d7d4941c0c84d2e60d0f8496e):

http://loganbuildbot.s3-website-us-east-1.amazonaws.com/

It still crashes if you are running using retroarch_debug.exe, but mupen64plus-libretro does the same thing so I am blaming RetroArch for that.

**Comment by SigmaVirus** [comment]
Sweet! Can't wait to try this out when I get home.
Thank you once again!
